docs/src/main/asciidoc/kafka.adoc

```diff
@@ -566,6 +568,47 @@ mp.messaging.incoming.your-channel.group.id=${quarkus.uuid}
 
 IMPORTANT: If the `group.id` attribute is not set, it defaults the `quarkus.application.name` configuration property.
 
+=== Receiving Kafka Records in Batches
+
+By default, incoming methods receive each Kafka record individually.
+Under the hood, Kafka consumer clients poll the broker constantly and receive records in batches, presented inside the `ConsumerRecords` container.
+
+In *batch* mode, your application can receive all the records returned by the consumer *poll* in one go.
+
+To achieve this you need to set `mp.messaging.incoming.$channel.batch=true` *and* specify a compatible container type to receive all the data:
+
+[source, java]
+----
+@Incoming("prices")
+public void consume(List<Double> prices) {
+    for (double price : prices) {
+        // process price
+    }
+}
+----
+
+The incoming method can also receive `Message<List<Payload>`, `KafkaBatchRecords<Payload>` `ConsumerRecords<Key, Payload>` types.
```

Comment:
Is it `KafkaBatchRecords` (mentioned here) or `KafkaRecordBatch` (mentioned in the example below)?

Reply:
Also:

```suggestion
The incoming method can also receive `Message<List<Payload>>`, `KafkaBatchRecords<Payload>` `ConsumerRecords<Key, Payload>` types.
```